components/style/build_gecko.rs

```diff
@@ -232,6 +232,16 @@ mod bindings {
         File::create(&out_file).unwrap().write_all(&result.into_bytes()).expect("Unable to write output");
     }
 
+    fn get_arc_types() -> Vec<String> {
+        let list_file = File::open(DISTDIR_PATH.join("include/mozilla/ServoArcTypeList.h"))
+            .expect("Unable to open ServoArcTypeList.h");
+        let re = Regex::new(r#"^SERVO_ARC_TYPE\(\w+,\s*(\w+)\)"#).unwrap();
```

Comment:
If we're not processing this using the C preprocessor, then please add some comments to ServoArcTypeList.h saying that the format of that file must be kept very strict, and that e.g. comments and #ifdefs will not be processed when the file is read for bindgen's purpose.

In fact, I wonder if you can cause the build to break in here if the file contents are unexpected.  Maybe you could allow only `//` comments, blank lines, and `SERVO_ARC_TYPE(..., ...)` lines, and panic if we get some other line?